### PR TITLE
Extend unittests for some distributions.

### LINKED ISSUE #33: Add test to check correctness of samples from multivariate normal distributions
We currently only check that the shape of the output is the expected shape, but we should at least make sure that the value of the mean & covariance of the samples are correct.

### PR DESCRIPTION
**Pull request by zoj613**
This commit fixes/completes/extends previously minimal unittests for
some of the available probability distribution samplers, which include:
- `multivariate_normal_cong2017`
- `multivariate_normal_bhattacharya2016`.

This is a followup of https://github.com/aesara-devs/aemcmc/pull/32 and closes #33

### CODE REVIEW

**Comment by zoj613** [comment]
For what its worth: the `multivariate_normal_bhattacharya2016` function is a special case of `multivariate_normal_cong2017` for when the `omega` parameter is the identity matrix, so it can be written as:
```python

def multivariate_normal_bhattacharya2016(rng, D, phi, alpha):
    return multivariate_normal_cong2017(rng, 1 / D, at.ones(phi.shape[0]), phi, alpha)
```
So not sure if this is worth a refactor and that the overhead of creating the `at.ones` and the extra steps inside the `multivariate_normal_cong2017` function wont affect performance.

Cc @rlouf @brandonwillard

**Comment by zoj613** [comment]
BTW @rlouf while working on this PR today I realized that the tests in #32 previously failed for two reasons:
1) you were using passing `phi` as a transpose of the expected matrix in the cong2017 function. Also the `phi` need for the other one should be computed from the cholesky since omega is the identity (so the eigen decomposition was not going to work since it produces a diagonal matrix that isnt the identity).
2) the true variance should have been the diagonal entries of the covariance matrix of the input data so that the samples from the `scan` function can be correctly compared to those entries.

**Comment by rlouf** [comment]
Thank you! I have not seen `multivariate_normal_bhattacharya2016` used anywhere. If so we should remove it altogether.

For the documentation it was not always easy to find the corresponding parametrization in the references. A short explanation of the meaning of each parameter and how they relate to mean/covariance would be very useful.

Besides these two point this looks good to me.

**Comment by zoj613** [comment]
> Thank you! I have not seen `multivariate_normal_bhattacharya2016` used anywhere. If so we should remove it altogether.
> 
> For the documentation it was not always easy to find the corresponding parametrization in the references. A short explanation of the meaning of each parameter and how they relate to mean/covariance would be very useful.
> 
> Besides these two point this looks good to me.

Do you feel these two points should be a separate PR or part of this one?

**Comment by rlouf** [comment]
They can be part of this one; it will take you more time to open a separate PR than make these changes.

**Comment by zoj613** [comment]
@rlouf  i've addressed your last 2 points in the latest commit. Please take a look at this again when you can,.

**Comment by rlouf** [comment]
The changes look good to me, however you will need to make some changes to your commits' structure:
1. No need for TST/DOC, etc in the commit title
2. Can you break down you last commit in two: removing one function + adding some documentation? These are two separate changes.

File: aemcmc/dists.py [hunk 1]
```diff
@@ -50,30 +52,93 @@ def multivariate_normal_rue2005(rng, b, Q):
...
+    t: TensorVariable
+        A 1D array whose length is the number of eigenvalues of the low-rank
+        part of the structured covariance.
+
+    Notes
+    -----
+    This algorithm is suitable for high-dimensional regression problems and the
+    runtime scales linearly with the number of regression coefficients. This
+    implementation assumes that `A` and `omega` are diagonal matrices and
+    the parameters `A` and ``omega`` are expected to be 1-dimension containing
```
Comment (rlouf) [inline aemcmc/dists.py:108]: "are expected to be vectors that contain the diagonal entries of the respective matrices they represent"

**Review by rlouf — APPROVED** [review]
Looks good to me.

File: aemcmc/dists.py [hunk 2]
```diff
@@ -50,30 +52,93 @@ def multivariate_normal_rue2005(rng, b, Q):
...
-    algorithm is suitable for high-dimensional regression problems and the
-    runtime scales linearly with the number of regression coefficients.
+def multivariate_normal_cong2017(
+    rng: RandomStream,
+    A: TensorVariable,
+    omega: TensorVariable,
+    phi: TensorVariable,
+    t: TensorVariable,
+) -> TensorVariable:
+    r"""
```
Comment (rlouf) [inline aemcmc/dists.py:62]: Nitpick: docstring should start after the `"""`

Comment (zoj613) [reply aemcmc/dists.py:62]: I think this is mainly a preference thing. The current way is still correct according the `numpydoc` convention and is used all over numpy and scipy source code.

Comment (rlouf) [reply aemcmc/dists.py:62]: When you start a repo every reasonable choice is certainly a question of *preference*, in an existing codebase it is a matter of *consistency*.

Comment (zoj613) [reply aemcmc/dists.py:62]: I understand your concern but `dists.py` is the first module commited into the codebase and if you look at the other functions in it, they use the same docstring convention. That is about as consistent as can be in this case. Using the suggested preference would introduce an inconsistent docstring style in the module. So the point made here is a bit contradictory.

Comment (rlouf) [reply aemcmc/dists.py:62]: There are only two choices here: change the docstrings in every other files in the repository, or press DEL twice in this file. Whatever you think is best.

**Comment by brandonwillard** [comment]
FYI: the `aesara-devs` repo doesn't use the "<tag>: <description>." PR title format.  (This is only worth mentioning because we generate our release descriptions using PR titles.)

File: aemcmc/dists.py (hunk 2 above)
Comment (brandonwillard) [reply aemcmc/dists.py:62]: All the `aesara-devs` projects follow [the code and documentation standards of Aesara](https://aesara.readthedocs.io/en/latest/dev_start_guide.html#documentation-and-docstrings).  In the case of docstrings, we use NumPy's format&mdash;more or less.  (The link in our documentation is broken; it should be [this](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard).)

We have some legacy docstrings in the Aesara repo that go against this standard, but those need to be fixed.

**Pull request merged**